### ISSUE TITLE
can not create header object from certain inputs

### ISSUE THREAD

**Issue by glensc** [zendframework/zend-mail#64]
When trying to load raw (QP encoded, etc) headers with:

```php
Zend\Mail\Headers::fromString($headerString);
```

I came up to a problem that [iconv_mime_encode](http://php.net/iconv_mime_encode) can not be used reliably to validate if header can be encoded

for example (the value must be exactly like that):
```php
        $name = 'Subject';
        $value = "[#77675] New Issue:xxxxxxxxx xxxxxxx xxxxxxxx xxxxxxxxxxxxx xxxxxxxxxx xxxxxxxx, tähtaeg xx.xx, xxxx";
        $header = new GenericHeader($name, $value);
```

the above code breaks at [HeaderWrap::canBeEncoded](https://github.com/zendframework/zend-mail/blob/release-2.5.2/src/Header/HeaderWrap.php#L119) when underlying `iconv_mime_encode` attempts to wordwrap the string and fails as it needs to wrap at middle of character or something:

```
iconv_mime_encode(): Unknown error (7)
```

which results `false` as return value and [InvalidArgumentException](https://github.com/zendframework/zend-mail/blob/release-2.5.2/src/Header/GenericHeader.php#L130-L132) later

The header will be encoded ok if the `$value` is lengthened or shortened even one byte before 76th character.

**Comment by glensc** [comment]
problem can be mitigated by specifying long enough `line-length` option.

i've [solved](https://github.com/eventum/eventum/blob/v3.0.9/lib/eventum/class.mime_helper.php#L221-L233)  this in the past by calculating what the length may be. but current zend code doesn't even have charset parameters, so it could be anything whatever is current php version and php.ini settings.

so, perhaps hardcode UTF-8 in that specific method and solve it similar way? i'll send PR (with test included) when i get some feedback.

also i'm interested having this on 2.4 branch as well, i read [LTS-WORKFLOW](https://github.com/zendframework/maintainers/blob/master/LTS-WORKFLOW.md) and seems the PR should be sent to zf2 repo for that? or maintainers will backport themselves? (the differences 2.4 vs 2.5 are rather in-existent, some array syntax changes only)

**Comment by glensc** [comment]
alternative would be to split the `$value` on spaces and try to encode each component, as say the [comments](http://php.net/iconv_mime_encode) on php.net for the function.

**Comment by glensc** [comment]
hallo?

**Comment by smazurov** [comment]
same issue as #33? we ran into this ourselves.

**Comment by glensc** [comment]
looks same to me

**Comment by glensc** [comment]
this case i even have bugfix, but no maintainer bothered to reply here, so never fired off the pull request

the branch itself is here:
https://github.com/zendframework/zend-mail/compare/master...glensc:issue-64

**Comment by ezimuel** [comment]
@glensc if you want to submit a PR for this issue it will be awesome! I just merged yours #69 and #61. Sorry again for the delay of my reply :(

**Comment by weierophinney** [comment]
> i'm interested having this on 2.4 branch as well, i read LTS-WORKFLOW and seems the PR should be sent to zf2 repo for that? or maintainers will backport themselves?

No; we determine if a patch is a candidate, and backport it ourselves, as we have a very specific workflow for that.

Right now, we're very conservative in what we backport, as the process is somewhat time-consuming, as well as brittle. Typical candidates are security fixes, and fixes for BC breaks introduced between 2.3 and 2.4. As such, this does not qualify.
